### DISCUSSION TITLE
Adding an OpenID Connect auth provider

### DISCUSSION THREAD

**Discussion by SteelAlloy** [mediacms-io/mediacms#693] [Q&A] [answered]
Hi,

Thanks for this great project !

I can't figure out how to add an OpenID Connect auth provider, there's almost no documentation on adding providers of any kind.
I know this project uses https://django-allauth.readthedocs.io/en/latest/providers.html#openid-connect but I have absolutely no idea where to put this config.

```python
SOCIALACCOUNT_PROVIDERS = {
    "openid_connect": {
        "SERVERS": [
            {
                "id": "my-server",  # 30 characters or less
                "name": "My Login Server",
                "server_url": "https://my.server.example.com",
                # Optional token endpoint authentication method.
                # May be one of "client_secret_basic", "client_secret_post"
                # If omitted, a method from the the server's
                # token auth methods list is used
                "token_auth_method": "client_secret_basic",
                "APP": {
                    "client_id": "your.service.id",
                    "secret": "your.service.secret",
                },
            },
            {
                "id": "other-server",  # 30 characters or less
                "name": "Other Login Server",
                "server_url": "https://other.server.example.com",
                "APP": {
                    "client_id": "your.other.service.id",
                    "secret": "your.other.service.secret",
                },
            },
        ]
    }
}
```

The admin panel is also very confusing, I don't know which panel does what...
![image](https://user-images.githubusercontent.com/52361520/211794766-38da2bc5-12e1-45b9-8052-5242b9caacc7.png)

Any help is appreciated ! 🙏

**Comment by SteelAlloy** [comment, marked as answer]
Solved partially thanks to #565 

EDIT : change `django-allauth` version to 0.52.0 otherwise `openid_connect` won't be available

**Reply by MAXimized490** [reply]
Hey, how exactly did you change the allauth version being used? Did you simply `pip install` that version? I'm running it through Docker and I'm running `pip install django-allauth==65.0.2` which seems to work but it still complains about the middleware module not existing. It doesn't complain and starts if I remove the middleware line but then it doesn't show the social login buttons.

Edit:
Sorry, I just realized it pulls the versions from the requirements.txt file. I tried setting this to a few different versions, including 0.52.0 and 65.0.2, and I'm still getting this error:
`ModuleNotFoundError: No module named 'allauth.account.middleware'`
